components/file_system/src/thread_io.rs

```diff
@@ -0,0 +1,258 @@
+        Arc::new(ThreadIOSentinel::current_default())
+    ).clone();
+
+    static THREAD_IO_PRIVATE: Cell<ThreadIOPrivate> = Cell::new(ThreadIOPrivate::default())
+}
+
+struct ThreadIOSentinel {
+    id: ThreadID,
+    bytes: [AtomicIOBytes; IOType::COUNT],
+}
+
+impl ThreadIOSentinel {
+    fn current_default() -> ThreadIOSentinel {
+        ThreadIOSentinel {
+            id: ThreadID::current(),
+            bytes: Default::default(),
+        }
+    }
+}
+
+struct ThreadID {
+    pid: Pid,
+    tid: Pid,
+}
+
+impl ThreadID {
+    fn current() -> ThreadID {
+        ThreadID {
+            pid: getpid(),
+            tid: gettid(),
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+struct ThreadIOPrivate {
+    io_type: IOType,
+    last_bytes: IOBytes,
+}
+
+struct AtomicIOBytes {
+    read: AtomicU64,
+    write: AtomicU64,
+}
+
+impl Default for AtomicIOBytes {
+    fn default() -> Self {
+        let bytes = IOBytes::default();
+
+        AtomicIOBytes {
+            read: AtomicU64::new(bytes.read),
+            write: AtomicU64::new(bytes.write),
+        }
+    }
+}
+
+impl AtomicIOBytes {
+    fn load(&self, order: Ordering) -> IOBytes {
+        IOBytes {
+            read: self.read.load(order),
+            write: self.write.load(order),
+        }
+    }
+
+    fn store(&self, val: IOBytes, order: Ordering) {
+        self.read.store(val.read, order);
+        self.write.store(val.write, order);
+    }
+
+    fn fetch_add(&self, other: IOBytes, order: Ordering) {
+        self.read.fetch_add(other.read, order);
+        self.write.fetch_add(other.write, order);
+    }
+}
+
+pub fn fetch_all_thread_io_bytes(io_type: IOType) -> IOBytes {
+    THREAD_IO_SENTINEL_VEC.iter().for_each(|sentinel| {
+        flush_sentinel(sentinel, None);
+    });
+    THREAD_IO_TOTAL[io_type as usize].load(Ordering::Relaxed)
+}
+
+fn flush_sentinel(sentinel: &ThreadIOSentinel, new_io_type: Option<IOType>) {
+    let io_bytes = fetch_exact_thread_io_bytes(&sentinel.id);
+
+    THREAD_IO_PRIVATE.with(|private_cell| {
+        let mut private = private_cell.get();
+        let io_type = private.io_type;
+        let last_io_bytes = private.last_bytes;
+
+        sentinel.bytes[io_type as usize].fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
+        THREAD_IO_TOTAL[io_type as usize].fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
+        private.last_bytes = io_bytes;
+        if let Some(new_io_type) = new_io_type {
+            private.io_type = new_io_type;
+        }
+        private_cell.set(private)
+    })
+}
+
+pub fn set_io_type(new_io_type: IOType) {
+    THREAD_IO_SENTINEL.with(|sentinel| {
+        flush_sentinel(sentinel, Some(new_io_type));
+    })
+}
+
+pub fn get_io_type() -> IOType {
+    THREAD_IO_PRIVATE.with(|private| private.get().io_type)
+}
+
+fn fetch_thread_io_bytes(_io_type: IOType) -> IOBytes {
+    fetch_exact_thread_io_bytes(&ThreadID::current())
+}
+
+fn fetch_exact_thread_io_bytes(id: &ThreadID) -> IOBytes {
```

Comment:
How about reuse `ThreadInspector` in `components/tikv_util/src/sys/inspector.rs`?

Reply:
I agree. The merged version needs to stay at "file_system" though, to avoid circular dependency.

Reply:
It seems `ThreadInspector` can only get `io_stat` of **current** thread now. But we need function to get `io_stat` of other thread here.